### PR TITLE
Fix auto using incorrect position for key up frames

### LINKED ISSUE #9341: Auto should have smoother movement when objects are close together in time
"smoother movement" as in more linear instead of snapping to each note

streams look really wrong on Auto right now. I think this is an easy way to make Auto look a lot more natural

### PR DESCRIPTION
**Pull request by PercyDan54**
Fixes #9341.
Opened a new PR since #11764 is closed(for inactivity?).

This time I tried to make inline comments more clear

Tested with:
Normal ranked maps (can SS):
https://osu.ppy.sh/beatmapsets/767600#osu/1613687
https://osu.ppy.sh/beatmapsets/1384189#osu/2859822
https://osu.ppy.sh/beatmapsets/292301#osu/658127

2B maps(no miss):
https://osu.ppy.sh/beatmapsets/1236988#osu/2573493
https://osu.ppy.sh/beatmapsets/1139945#osu/2381064

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Replays/OsuAutoGenerator.cs [hunk 1]
```diff
@@ -22,6 +22,8 @@ namespace osu.Game.Rulesets.Osu.Replays
     public class OsuAutoGenerator : OsuAutoGeneratorBase
     {
         public new OsuBeatmap Beatmap => (OsuBeatmap)base.Beatmap;
+        private readonly List<OsuReplayFrame> framesPendingAdjust = new List<OsuReplayFrame>();
```
Comment (PercyDan54) [inline osu.Game.Rulesets.Osu/Replays/OsuAutoGenerator.cs:25]: I think we need a better name

**Comment by smoogipoo** [comment]
As I said in the original PR:

> This is definitely something we'll want, but at this point we're all having trouble wrapping our heads around this. A large part of the complexity is that the "adjustments" are interleaved through the actual generation process, and I personally would like to see this split into multiple passes in order to simplify this.

You've basically not changed this PR at all, and it's going to get stuck on the same thing. Perhaps you can understand it because you wrote it, but for anyone else it's a minimum one-day effort to understand 40 LOC and to verify that simple concepts like ordering frames by strictly increasing times, as a result of the `framesAfterAdjust.ForEach(AddFrameToReplay)` post-op, are adhered to.

This process needs to be split out into multiple passes, for me to review this PR effectively.

**Comment by PercyDan54** [comment]
> As I said in the original PR:
> 
> > This is definitely something we'll want, but at this point we're all having trouble wrapping our heads around this. A large part of the complexity is that the "adjustments" are interleaved through the actual generation process, and I personally would like to see this split into multiple passes in order to simplify this.
> 
> You've basically not changed this PR at all, and it's going to get stuck on the same thing. Perhaps you can understand it because you wrote it, but for anyone else it's a minimum one-day effort to understand 40 LOC and to verify that simple concepts like ordering frames by strictly increasing times, as a result of the `framesAfterAdjust.ForEach(AddFrameToReplay)` post-op, are adhered to.
> 
> This process needs to be split out into multiple passes, for me to review this PR effectively.

I did not add subprocess for this because I don't want another iteration through hitobjects, so all process is done in the original order

**Comment by smoogipoo** [comment]
Do you have a strong reason for that? O(n) performance is not a concern.

**Comment by frenzibyte** [comment]
Are you still looking into this? or is this gonna be left behind like its predecessor.

**Comment by PercyDan54** [comment]
> Are you still looking into this? or is this gonna be left behind like its predecessor.

Yes.
And I had no idea how to split this further

**Comment by bdach** [comment]
Let me just state this: For me personally to even approach this pull from a review standpoint, *all* of the added inline comments *need* to be torn out and replaced with an *actual* explanation. Not one of them helps in the current state. They are all cryptic, because they mostly say what is being done and not why, and this therefore diff is inscrutable to me.

As things stand I can only tell that this makes "autoplay look nicer" but I have zero clue as to how it achieves that. And as stated above I am not going to spend a day to reverse-engineer it myself. Because it's already written and I expect you, the author, to be able to explain how it works in depth.

**Pull request closed without merging**